kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/dates.kt

```diff
@@ -42,31 +43,47 @@ fun Arb.Companion.period(maxYear: Int = 10): Arb<Period> = arbitrary(listOf(Peri
 fun Arb.Companion.localDate() = Arb.Companion.localDate(LocalDate.of(1970, 1, 1), LocalDate.of(2030, 12, 31))
 
 /**
- * Arberates a stream of random LocalDates
+ * Generates a stream of random [LocalDate] instances within the specified range.
  *
- * This generator creates randomly generated LocalDates, in the range [[minDate, maxDate]].
+ * This generator creates random [LocalDate] values in the inclusive range from [minDate] to [maxDate].
  *
- * If any of the years in the range contain a leap year, the date [29/02/YEAR] will always be a constant value of this
- * generator.
+ * It includes special edge cases for testing purposes:
+ * - If the range includes any leap years, the date **February 29th** of the first such leap year is included.
+ * - If the range includes any century years (years divisible by 100), the date **January 1st** of the first such century year is included.
+ * - If the range includes any millennium years (years divisible by 1000), the date **January 1st** of the first such millennium year is included.
  *
- * @see [localDateTime]
- * @see [localTime]
+ * These edge cases are added to increase the likelihood of covering date-related boundary conditions in tests.
+ *
+ * @param minDate The minimum (earliest) date to generate (inclusive). Default is January 1st, 1970.
+ * @param maxDate The maximum (latest) date to generate (inclusive). Default is December 31st, 2030.
+ * @return An [Arb]<[LocalDate]> that generates dates within the specified range, including edge cases.
+ *
+ * @see localDateTime
+ * @see localTime
  */
 fun Arb.Companion.localDate(
    minDate: LocalDate = LocalDate.of(1970, 1, 1),
    maxDate: LocalDate = LocalDate.of(2030, 12, 31)
-): Arb<LocalDate> = when {
-   minDate > maxDate -> throw IllegalArgumentException("minDate must be before or equal to maxDate")
-   minDate == maxDate -> Arb.constant(minDate)
-   else -> {
-      val leapYears = (minDate.year..maxDate.year).filter { isLeap(it.toLong()) }
+): Arb<LocalDate> {
+   require(minDate <= maxDate) { "minDate must be before or equal to maxDate" }
+   if (minDate == maxDate) return Arb.constant(minDate)
+
+   val edgeCases = mutableListOf(minDate, maxDate)
+
+   val leapYear = (minDate.year..maxDate.year).firstOrNull { isLeap(it.toLong()) }
```

Comment:
Let's modify it to 
`val leapYear = (minDate.year..maxDate.year).firstOrNull { isLeap(it.toLong()) && LocalDate.of(it, 2, 29) >= minDate }`
so that we don't skip this edge case if, for instance, `minDate=LocalDate.of(2024, 3, 1)`

Do we need a unit test for this scenario?

The same goes for two other edge cases - we need to make sure edge cases are inside the range, right in the filter

Reply:
This specific scenario is covered out by having the `removeAll` clause.

>  so that we don't skip this edge case if, for instance, minDate=LocalDate.of(2024, 3, 1)

I don't get this example, as 2024-02-29 is indeed less than 2024-03-01. Could you clarify?

Reply:
Suppose the range is as follows:
```kotlin
 val minDate: LocalDate = LocalDate.of(2024, 3, 1),
  val maxDate: LocalDate = LocalDate.of(2030, 12, 31)
```
the way I've grokked the PR, this means that with the code as is `leapYear == 2024`, and later `2024-02-29` is filtered out by `edgeCases.removeAll { it !in minDate..maxDate }`.

This is why I'm suggesting  a change that will compute `leapYear == 2028` and include `2028-02-29` as the edge case. Does it make sense to you?

Reply:
Likewise, if the range is "2000-01-02".."2100-01-02", the centureYear edge case of "2100-01-01" is in the range., but "2000-01-01" is not. Should we provide "2100-01-01"? Please correct me if I am wrong.

Reply:
Ah, I see.
That's crystal clear now! I'll adjust the code

Reply:
Done